packages/mobile/src/language/Language.tsx

```diff
@@ -32,18 +32,14 @@ function keyExtractor(item: Language) {
 function LanguageScreen({ route }: Props) {
   const dispatch = useDispatch()
   const { t, i18n } = useTranslation(Namespaces.accountScreen10)
-  const fromSettings = route.params?.fromSettings
+  const nextScreen = route.params?.nextScreen
 
   const onSelect = (language: string, code: string) => {
     dispatch(setLanguage(code))
-    // Wait for next frame before navigating back
+    // Wait for next frame before navigating
     // so the user can see the changed selection briefly
     requestAnimationFrame(() => {
-      if (fromSettings) {
-        navigateBack()
-      } else {
-        navigate(Screens.JoinCelo)
-      }
+      navigate(nextScreen || Screens.OnboardingEducationScreen)
```

Comment:
Can you please confirm that this works as expected when accessing language through Settings? When we were previously pushing to the stack rather than popping off the stack, it was causing undesirable results (e.g., you'd need to hit back button twice or top bar navigator would become unresponsive)